q2_makarsa/plugin_setup.py

```diff
@@ -107,3 +108,42 @@ def _2(ff: NetworkFormat) -> Graph:
         "This method generates the sparse matrix of network of input " "data"
     ),
 )
+
+plugin.register_semantic_types(Node_Community)
+plugin.register_formats(NodeCommunityFormat, NodeDirectoryFormat)
+plugin.register_semantic_type_to_format(
+    Node_Community, artifact_format=NodeDirectoryFormat
+)
+
+
+
+
+@plugin.register_transformer
```

Comment:
you also need a transformer to go from `NodeCommunityFormat` to `pd.DataFrame` and from `NodeCommunityFormat` to `qiime2.Metadata`. The latter will allow the `NodeCommunityFormat` artifact to be passed to other existing actions in QIIME2, e.g., for collapsing a table into modules.